src/sfizz/Voice.cpp

```diff
@@ -718,3 +734,24 @@ void sfz::Voice::setupOscillatorUnison()
     }
 #endif
 }
+
+
+void sfz::Voice::updateChannelPowers(AudioSpan<float> buffer)
+{
+    assert(channelPowers.size() == channelPowerFilters.size());
+    assert(buffer.getNumFrames() <= channelPowerFilters.size());
+    if (buffer.getNumFrames() == 0)
+        return;
+
+    auto tempSpan = resources.bufferPool.getBuffer(buffer.getNumFrames());
+    if (!tempSpan)
+        return;
+
+    for (unsigned i = 0; i < channelPowers.size(); ++i) {
+        for (unsigned s = 0; s < buffer.getNumFrames(); ++s)
+            (*tempSpan)[s] = std::abs(buffer.getConstSpan(i)[s]);
+
+        channelPowerFilters[i].processLowpass(*tempSpan, *tempSpan);
```

Comment:
Can the one-pole filter have:
- `inline float tickLowpass(float x)`
- `inline float tickHighpass(float x)`
There isn't benefit to run it in buffers, since it's data-dependent from one cycle to the next.
This eliminates the temp span.

Reply:
Done :+1: